src/lsm/lsm_tree.c

```diff
@@ -1385,6 +1358,22 @@ __wt_lsm_tree_worker(WT_SESSION_IMPL *session,
 			WT_ERR(__wt_schema_worker(session, chunk->bloom_uri,
 			    file_func, name_func, cfg, open_flags));
 	}
+	/*
+	 * If this was an alter operation, we need to alter the configuration
+	 * for the overall tree and then reread it so it isn't out of date.
+	 * Reread it here so that we update the configuration of the
+	 * current tree's structure to any new, altered values.
+	 */
+	if (FLD_ISSET(open_flags, WT_BTREE_ALTER)) {
+		WT_ERR(__wt_lsm_meta_write(session, lsm_tree, cfg[0]));
+		/*
+		 * We're about to read in the new configuration that
+		 * we just wrote.  Free the old ones.
+		 */
+		__lsm_tree_meta_discard(session, lsm_tree);
+		WT_ERR(__wt_lsm_meta_read(session, lsm_tree));
```

Comment:
Any error from here must need to be fatal - right? Otherwise the in-memory tree structure will be invalid.